src/mouse.py

```diff
@@ -1,60 +1,105 @@
 
         self.maze.grid[self.row][self.column] = 2
 
-    def move(
-        self, direction
-    ):  # This function is how a character moves around in a certain direction
-        {
-            "UP": self.move_up,
-            "DOWN": self.move_down,
-            "LEFT": self.move_left,
-            "RIGHT": self.move_right,
+    def move(self, direction) -> int:
+        """
+        This function is how a character moves around in a certain direction
+        """
+        return {
+            self.Moves.UP: self.move_up,
+            self.Moves.DOWN: self.move_down,
+            self.Moves.LEFT: self.move_left,
+            self.Moves.RIGHT: self.move_right,
         }[direction]()
 
-    def move_up(self):
-        if self.row > 0 and not self.collision("UP"):
+    def move_up(self) -> int:
+        if self.row > 0 and not self.collision(self.Moves.UP):
             self.maze.grid[self.row][self.column] = 0
             self.row -= 1
+
+            self.end_reached()
+
             self.maze.grid[self.row][self.column] = 2
+            return 5
+        return -5
 
-    def move_down(self):
-        if self.row < self.maze.height - 1 and not self.collision("DOWN"):
+    def move_down(self) -> int:
+        if self.row < self.maze.height - 1 and not self.collision(self.Moves.DOWN):
             self.maze.grid[self.row][self.column] = 0
             self.row += 1
+
+            self.end_reached()
+
             self.maze.grid[self.row][self.column] = 2
+            return 5
+        return -5
 
-    def move_left(self):
-        if self.column > 0 and not self.collision("LEFT"):
+    def move_left(self) -> int:
+        if self.column > 0 and not self.collision(self.Moves.LEFT):
             self.maze.grid[self.row][self.column] = 0
             self.column -= 1
+
+            self.end_reached()
+
             self.maze.grid[self.row][self.column] = 2
+            return 5
+        return -5
 
-    def move_right(self):
-        if self.column < self.maze.width - 1 and not self.collision("RIGHT"):
+    def move_right(self) -> int:
+        if self.column < self.maze.width - 1 and not self.collision(self.Moves.RIGHT):
             self.maze.grid[self.row][self.column] = 0
             self.column += 1
+
+            self.end_reached()
+
             self.maze.grid[self.row][self.column] = 2
+            return 5
+        return -5
 
-    def collision(self, direction) -> bool:
+    def collision(self, direction: Moves) -> bool:
         """
         Checks if anything is on top of the grass in the direction
         that the character wants to move. Used in the move function
         """
-        if direction == "UP":
+        if direction == self.Moves.UP:
             if self.maze.grid[self.row - 1][self.column] != 0:
                 return True
-        elif direction == "LEFT":
+        elif direction == self.Moves.LEFT:
             if self.maze.grid[self.row][self.column - 1] != 0:
                 return True
```

Comment:
```suggestion
            return self.maze.grid[self.row][self.column - 1] == 1
```
maybe this way and then return False at the bottom could be removed